src/wms/src/main/java/org/geoserver/wms/WMS.java

```diff
@@ -1253,6 +1260,42 @@ private String sortSpecification(SortBy sb) {
         return queryFeatureTypeDimension(typeInfo, range, maxItems, ResourceInfo.TIME);
     }
 
+    /** Returns a {@link SimpleDateFormat} using the UTC_PATTERN and the UTC time zone */
+    public SimpleDateFormat getTimeFormat() {
+        final SimpleDateFormat df = new SimpleDateFormat(UTC_PATTERN);
+        df.setTimeZone(UTC_TIME_ZONE);
+        return df;
+    }
+
+    /**
+     * Returns available fixed values for a Time dimension.
+     *
+     * @param spec info that holds the Time dimension.
+     * @return fixed values date type list.
+     */
+    public TreeSet<Date> getFixedValueRangeTimes(String spec) {
+        TreeSet<Date> result = new TreeSet<>();
+        SimpleDateFormat outputDateFormat = getTimeFormat();
+        if (spec == null || spec.trim().isEmpty()) {
+            return null;
+        }
+
+        List<String> fixedValueList = Arrays.asList(spec.split(","));
```

Comment:
We should use the DateTImeParser not a fixed date format. Also I would change the method name to parseFixedValues.